data/pools/transactionPool.go

```diff
@@ -509,20 +514,22 @@ func (pool *TransactionPool) recomputeBlockEvaluator(committedTxIds map[transact
 		return
 	}
 
+	// Grab the transactions to be played through the new block evaluator
+	pool.pendingMu.RLock()
+	txgroups := pool.pendingTxGroups
+	verifyParams := pool.pendingVerifyParams
+	pendingCount := pool.pendingCountNoLock()
+	pool.pendingMu.RUnlock()
+
 	next := bookkeeping.MakeBlock(prev)
 	pool.numPendingWholeBlocks = 0
-	pool.pendingBlockEvaluator, err = pool.ledger.StartEvaluator(next.BlockHeader)
+	pool.pendingBlockEvaluator, err = pool.ledger.StartEvaluator(next.BlockHeader, pendingCount)
```

Comment:
I'm curious why you chose to override the payset allocation inside StartEvaluator rather then when it was originally allocated in MakeBlock ( as nil, if I recall correctly ).
It's also weird that we create a new block ( i.e. MakeBlock ) just to steal it's header. but I guesses it's just semantics.

Reply:
Yeah, `StartEvaluator` takes in a block header and not a whole block, and changing that behavior seemed like a much bigger design change.